### DISCUSSION TITLE
INSTALLATION_METRICS FAILED

### DISCUSSION THREAD

**Discussion by AlvaroRaez** [nautobot/nautobot#4846] [Q&A]
Hello everyone,

Between yesterday and today, I have been searching for how to install metrics on my Nautobot. I find this plugin really useful for extracting all the data stored in the tool.

According to the Nautobot documentation (https://docs.nautobot.com/projects/capacity-metrics/en/latest/admin/install/#install-guide), I consider that the steps to follow are not entirely correct since there are several open cases regarding this.

To run the migrations using the command nautobot-server post_upgrade, it fails, as indicated by the open cases related to this issue.

During the installation of metrics at https://nautobot.cloud/api/nautobot/installation-metric, when I input the following code in Raw Data (Copy and paste), it fails:

![image](https://github.com/nautobot/nautobot/assets/150433719/a97d38c6-92e8-4ec1-86b3-ed41c7f01b8f)

Is there any way to install this plugin correctly? On the other hand, how is the JSON that needs to be stored to display the dashboard done? I am unaware of whether it needs to be stored in a specific path.

THANK YOU

**Comment by bryanculver** [comment]
@AlvaroRaez Does your environment use a proxy to get out to the public internet?

**Comment by lampwins** [comment]
@AlvaroRaez are you possibly talking about the [capacity metrics app](https://github.com/nautobot/nautobot-plugin-capacity-metrics)?

`INSTALLATION_METRICS` is a setting that deals with telemetry that we use to get an idea of the number of Nautobot instances deployed in the wild which helps us drive product direction. It doesn't have anything to do with application metrics exposed from the app. The reason you would see an error here is because your instance is not able to get out to the internet to reach the metrics reporting API endpoint. Possibly because of proxy issues as @bryanculver mentioned.

However, the capacity metrics app is not related to installation metrics in any way. There are also Prometheus metrics which can be enabled from the [configuration](https://docs.nautobot.com/projects/core/en/stable/user-guide/administration/guides/prometheus-metrics/). Maybe that also helps?

**Reply by lampwins** [reply]
Sorry, I see now that you are in fact talking about the capacity metrics app, from the installation link you provided. Tagging @Kircheneer as he might be able to assist with the install.

**Reply by Kircheneer** [reply]
If @lampwins is guessing correctly: After installing the capacity metrics app and activating it in the configuration under `PLUGINS` you should see a new API endpoint at `your.nautobot.installation/api/plugins/capacity-metrics/app-metrics` which contains the metrics that capacity metrics exposes. What kind of error/problem are you encountering in the installation process?

**Reply by Kircheneer** [reply]
For some indication of what kind of metrics this includes check here: https://demo.nautobot.com/api/plugins/capacity-metrics/app-metrics

**Reply by AlvaroRaez** [reply]
Hi  @lampwins @Kircheneer @bryanculver 


First of all, I want to thank you for your prompt response.

I believe it's necessary to provide a bit more context. In my tool, I have more than 400 devices, 4 locations, 150 IPs, etc. These are quite relevant data as they represent an entire site that I manage.

![image](https://github.com/nautobot/nautobot/assets/150433719/b61f04e2-fb4c-48a7-ae42-30bc42c51139)


While investigating the repositories and plugins of Nautobot, I found that a plugin was released that allows visualizing almost all relevant records in my tool as images.

![image](https://github.com/nautobot/nautobot/assets/150433719/992ed0bd-6a45-49cc-9d17-7dc6f67d5c40)
![image](https://github.com/nautobot/nautobot/assets/150433719/cca127b5-1880-4fac-886e-dfa874249f74)

After reading the documentation at [GitHub](https://github.com/nautobot/nautobot-plugin-capacity-metrics) the error reported is as follows:

It failed after running the "nautobot-server post_upgrade" command:

![image](https://github.com/nautobot/nautobot/assets/150433719/6cbe105e-fcf5-4b48-958e-67d05ba23ac9)


 * Could it be an SSLCertVerificationError proxy error?
 * Is there something else I need to execute in the console?
 * Are the images found at (link) what the user will visualize somehow if they install the tool correctly?
 * Do we need to invoke/create new paths to visualize the new dashboard?

**Reply by Kircheneer** [reply]
As @bryanculver and @lampwins have alluded to, this error is not related to the capacity metrics plugin. You should be able to get rid of this message by either enabling your instance to connect to Nautobot cloud directly or by disabling the feature: https://docs.nautobot.com/projects/core/en/stable/user-guide/administration/configuration/optional-settings/?h=installation_metrics#installation_metrics_enabled

**Reply by whitej6** [reply]
It is likely you are behind a proxy and outbound internet traffic is hitting a proxy that's attempting to decrypt and reencrypt traffic using an internal certificate. The failure starting in 1.6.1 is no longer blocking startup and is informational, 1.6.0 is the only release where this could prevent startup. You can either set the http_proxy environment variable in your environment or disable installation metrics to get rid of the error being displayed.

This is due urllib3 and requests handling ssl verification and there not being a way to distinguish between an actual ssl error and an internet proxy intercepting the request.
